ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/transaction/ApiTransactionService.scala

```diff
@@ -134,8 +133,9 @@ final class ApiTransactionService private (
       logging.eventId(request.eventId),
       logging.parties(request.requestingParties),
     ) { implicit logCtx =>
-      EventIdFormatter
-        .split(request.eventId.unwrap)
+      TransactionIdWithIndex
+        .fromString(request.eventId.unwrap)
+        .toOption
```

Comment:
Why not keep the `Either[String, EventId]` and actually use the error message?

Reply:
done.